lib/screens/misc/inapp_webview_page.dart

```diff
@@ -39,26 +39,35 @@ class _WebViewWidgetState extends State<WebViewWidget> {
       walletAddress: widget.walletAddress,
       currency: 'GBP',
     );
-    Timer timer = Timer(Duration(seconds: 25), () {
-      Navigator.of(context, rootNavigator: true).pop();
-      showDialog(
-        context: context,
-        builder: (context) => TimedOut(),
-        barrierDismissible: true,
-      );
-    });
     if (response.ok) {
+      Segment.track(eventName: 'TopUp requested at Checkout');
+      Timer timer = Timer(Duration(seconds: 25), () {
+        Navigator.of(context, rootNavigator: true).pop();
+        Segment.track(eventName: 'User topup timed-out at checkout');
+        //Could cause issues with slow internet at checkout
+      });
       showDialog(
         context: context,
-        builder: (context) => MintingDialog(amount, false),
+        builder: (context) {
+          return MintingDialog(amount, false);
+        },
         barrierDismissible: false,
       ).then((value) {
```

Comment:
@Obinna-e When is this promise fulfilled (and the _then_ function called)?

Reply:
The timer was started when the if function's condition was fulfilled, so if the future(showdialog) doesn't move from MintingDialog widget to the TopUpSuccess widget within that time, the .then function gets executed

Reply:
Okay cool!